### DISCUSSION TITLE
Have doubt on the network connection

### DISCUSSION THREAD

**Discussion by chetu-ko** [WalletConnect/WalletConnectFlutterV2#276] [Q&A]
Hello, I am using SDK to connect the wallet through the app, I am connecting the wallet through the signEngine.connect for required and optional namespaces.

Here I have registerd the Mumbai testenet through the requiredNamespces but my wallet is connected with bsc network, When wallet connect popup occurs, I clicked on the connect It is conected the wallet, I got the two accounts one is for mumbai with wallet address another for the BSC. Can any one please explain why it's giving both, it should give me error of network miss match. Please correct me if I understood the concept wrong.

<img width="434" alt="Screenshot 2024-02-29 at 3 59 44 PM" src="https://github.com/WalletConnect/WalletConnectFlutterV2/assets/60283981/d0601742-9e02-4f23-ae6a-01985a80d4de">

Thank you.🙂

**Comment by quetool** [comment]
Hello @chetu-ko, can you share steps to reproduce?

**Reply by chetu-ko** [reply]
Below for the connection 
```
ConnectResponse resp =
        await web3App.signEngine.connect(requiredNamespaces: {
      'eip155': requiredNamespace1,
    }, optionalNamespaces: {
      'eip155': optionalNamespace,
    });

requiredNamespace1 = RequiredNamespace(
      chains: eip155,
      methods: kAppRequiredEIP155Methods,
      events: kAppRequiredEIP155Events,
    );

    optionalNamespace = RequiredNamespace(
      chains: eip155,
      methods: kAppOptionalEIP155Methods,
      events: kAppOptionalEIP155Events,
    );
```

These are the steps for the connection. Next, I am just navigating through the paring URI to a particular app. When I connected with the popup got the above two accounts.

**Reply by quetool** [reply]
Hello @chetu-ko! I'll need a fully reproducible code to debug, that means, I need exactly what you are requesting in requiredNamespaces and optionalNamespaces.

In any case, it is safe that whatever you want to request, you do it by just using optionalNamespaces. You can leave requiredNamespaces blank.

**Reply by chetu-ko** [reply]
@quetool Thank you for your reply. There is no error or misconception in the package, I need to understand the workflow and how it is happening. I just wanted to clear my doubts can you please explain why it's returning two addresses instead of one, The rest of the things in the packages are amazing🤩.

Below are the missing details for the above code.

```
const List<String> kAppRequiredEIP155Methods = [
  'personal_sign',
  'eth_sendTransaction',
];

const List<String> kAppOptionalEIP155Methods = [
  'signTypedData_v4', 
  'wallet_switchEthereumChain', 
];

const List<String> kAppRequiredEIP155Events =[
  'accountsChanged', 
  'chainChanged',
  'connect', 
  'disconnect', 
];
const List<String> kAppOptionalEIP155Events = [
  'accountsChanged', 
  'chainChanged', 
  'connect',
  'disconnect', 
];
```

**Reply by quetool** [reply]
What do you mean exactly by "2 addresses"? 2 different addresses or 2 accounts in the form of `eip155:1:0xSomeAddress`?

**Reply by chetu-ko** [reply]
@quetool 2 address of the different networks, no worries I think as I am passing account changed and chain changed event in the requiredEvents that's why changing network and accounts returning two addresses of the same account for the different network
